src/components/Editor/Runners/HtmlRunner/HtmlRunner.jsx

```diff
@@ -194,107 +202,105 @@ function HtmlRunner() {
     }
   }, [runningFile]);
 
-  const runCode = () => {
-    setRunningFile(previewFile);
-    if (!externalLink) {
-      let indexPage = parse(focussedComponent(previewFile).content);
-
-      const body = indexPage.querySelector("body") || indexPage;
+  const replaceHrefNodes = (indexPage, projectCode) => {
```

Comment:
💭 this looks pretty self contained now too, do you reckon this could be moved out to another helper? Thinking there's a few functions now related to manipulating the iframe `src` code that could be moved out

Reply:
nope, too tangled!